### PR TITLE
Prune vendor

### LINKED ISSUE #1722: Prune vendor directory?
When I migrated us to dep, I didn't turn on pruning because of a gap in dep, and how it treats the code-generator repo that caused dep to no properly vendor the non-go code in that repo. Once https://github.com/golang/dep/pull/1545 is merged however, we can turn pruning on if we like.

Do we want to?

### PR DESCRIPTION
**Pull request by carolynvs**
This removes about 8k files from vendor. 💃 🕺🏻

Closes #1722.

### CODE REVIEW

**Comment by MHBauer** [comment]
lol, somehow we picked up a copy of bazel

**Comment by carolynvs** [comment]
what the what?! dagnabbit one sec...

**Comment by MHBauer** [comment]
no, sorry, not because of this. I mean in the past we had.

**Comment by MHBauer** [comment]
This looks good so far, I'm just waiting for a build before I LGTM it.

**Comment by carolynvs** [comment]
Phew! 😅

**Comment by MHBauer** [comment]
@carolynvs dep seems unhappy.

**Comment by carolynvs** [comment]
Fixing the build, forgot to bump the version of dep in the build image...

**Comment by carolynvs** [comment]
Okay all fixed!

File: Gopkg.toml [hunk 1]
```diff
@@ -64,3 +64,17 @@ required = [
 [[constraint]]
   name = "k8s.io/code-generator"
   version = "kubernetes-1.9.1"
+
+[prune]
+  non-go = true
+  go-tests = true
+  unused-packages = true
```
Comment (MHBauer) [inline Gopkg.toml]: My understanding is that this says,

Do not include:
- non-go files
- go tests
- pkgs we don't build

**Comment by MHBauer** [comment]
Builds fine here. 

/LGTM

as an aside, I know it's all going to be squashed into one commit, (Which irks me to no end), but preferably 86d2dbbb7435235f915bcc398842439844612399 would be before e4ee5d8f8b32bb3a4d0ec56e68151ca9ba549ca6.

File: Gopkg.toml (hunk 1 above)
Comment (carolynvs) [reply Gopkg.toml]: Yes, with the exception of repo-infra because some of the code-gen related files are .txt files, etc and we don't want those nuked.

**Comment by carolynvs** [comment]
OK I'll rebase and flip the commit order

**Pull request merged**